gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -914,7 +954,11 @@ Builder setUseS2A(boolean useS2A) {
      */
     @InternalApi
     public Builder setAllowHardBoundTokenTypes(List<HardBoundTokenTypes> allowedValues) {
-      this.allowedHardBoundTokenTypes = allowedValues;
+      if (allowedValues == null) {
+        this.allowedHardBoundTokenTypes = new ArrayList<>();
+      } else {
+        this.allowedHardBoundTokenTypes = allowedValues;
+      }
```

Comment:
Can we use [Preconditions.checkNotNull](https://guava.dev/releases/19.0/api/docs/com/google/common/base/Preconditions.html#checkNotNull(T)) on this list?

And assign only if that passes

Reply:
Also I think you need to add `this.allowedHardBoundTokenTypes = ...`  to the Builder's constructor.

Reply:
Thanks for pointing these things out! Done in https://github.com/googleapis/sdk-platform-java/pull/3591/commits/d8b2959f1e05d4158570a2d9af93ff061ac8f310